app/controllers/users_controller.rb

```diff
@@ -0,0 +1,41 @@
+class UsersController < ApplicationController
+  skip_before_action :require_login
+  before_action :load_current_user, only: [:show, :update]
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      auto_login(@user)
+      redirect_to root_path
+      flash[:notice] = 'User was successfully created'
+    else
+      render 'new'
+    end
+  end
+
+  def show
+  end
+
+  def update
+    if (@user).update(user_params)
+      flash[:notice] = 'Update user successfully'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.